components/observations/AvalancheImagePicker.tsx

```diff
@@ -0,0 +1,242 @@
+import {AntDesign, MaterialIcons} from '@expo/vector-icons';
+import * as Sentry from '@sentry/react-native';
+
+import * as ImagePicker from 'expo-image-picker';
+import React, {useCallback, useState} from 'react';
+import {useController} from 'react-hook-form';
+import {ColorValue, StyleSheet, TouchableHighlight, View} from 'react-native';
+
+import {Button} from 'components/content/Button';
+import {NetworkImage} from 'components/content/carousel/NetworkImage';
+import {HStack, VStack, ViewProps} from 'components/core';
+import {AvalancheObservationFormData, ImageAndCaption} from 'components/observations/ObservationFormData';
+import {ImageCaptionField, ImageSizingView} from 'components/observations/ObservationImagePicker';
+import {getUploader} from 'components/observations/uploader/ObservationsUploader';
+import {Body, BodyBlack, BodySm} from 'components/text';
+import {LoggerContext, LoggerProps} from 'loggerContext';
+import Toast from 'react-native-toast-message';
+import {colorLookup} from 'theme';
+
+export const useAvalanchePickImages = ({maxImageCount, disable}: {maxImageCount: number; disable: boolean}) => {
+  const [imagePermissions] = ImagePicker.useMediaLibraryPermissions();
+  const missingImagePermissions = imagePermissions !== null && !imagePermissions.granted && !imagePermissions.canAskAgain;
+
+  const {field} = useController<AvalancheObservationFormData, 'images'>({name: 'images', defaultValue: []});
+  const images = field.value;
+  const imageCount = images?.length ?? 0;
+
+  const isDisabled = imageCount === maxImageCount || disable || missingImagePermissions;
+
+  const {logger} = React.useContext<LoggerProps>(LoggerContext);
+  const pickImage = useCallback(() => {
```

Comment:
(general comment) if these are common/shared logic between this picker and the other picker, let's see if we can refactor to DRY it out - if there are substantial differences that make that difficult, then let's not